### PR TITLE
added linux installation instructions

### PR DESCRIPTION
**Pull request by MisterRios**
I setup a virtual linux box on Digital Ocean and went through the mac instructions, but adapted them to linux.

Afterwards, I checked that Django was running by entering 

```
python manage.py runserver 0.0.0.0:8000 
```

(which broadcasts the django app to the whole network on port 8000) and then visiting the droplet's address with :8000 tacked to the end. It worked!

### CODE REVIEW

**Comment by shazow** [comment]
This is a dangerous thing to do. Django's built-in server is not built to be secure enough to broadcast to the outside world.

Deploying an app to the public is a different procedure (harder) than developing locally (easier). For deploying, there are some instructions about doing it with Heroku.

File: installation-instructions/python-installation-linux.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## Installing git
+
+The next thing we need to have installed is [git](http://git-scm.com/), which is
+a version control system. You can work on your programming projects while
+committing versions, allowing you to roll back easily if something you do
+breaks. git also works with GitHub, so you can save your project to the
+"cloud", backing it up as well as allowing others to work on the same project
+with you.
+
+Since we have Homebrew installed, it's easy to install git:
```
Comment (shazow) [inline installation-instructions/python-installation-linux.md]: Homebrew is not a thing on any Linux distro. :)

File: installation-instructions/python-installation-linux.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+Ubuntu and Fedora come with the latest versions of Python pre-installed, which means you don't need to install Python at all. However, [Setuptools](https://pypi.python.org/pypi/setuptools#unix-wget) is a fairly important Python package, which you should install before you do anything else. 
+
+```
+wget https://bootstrap.pypa.io/ez_setup.py -O - | python
+```
+
+if you get an error, you might have to invoke your sudo privileges to install:
+
+```
+wget https://bootstrap.pypa.io/ez_setup.py -O - | sudo python
```
Comment (shazow) [inline installation-instructions/python-installation-linux.md]: It's a bad habit to encourage people to pipe things off the internet into a shell, let alone a sudo shell. Probably best to use the distro's package manager to install setuptools if it doesn't come pre-installed.

File: installation-instructions/python-installation-linux.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+We'll go over some of these basics in the book, but if you'd like to do some
+advanced reading on git basics, [check out this
+guide](http://git-scm.com/book/en/Git-Basics).
+
+## Installing pip
+
+[pip](http://www.pip-installer.org/en/latest/) is essentially a package manager for Python packages. 
+```
+apt-get install python-pip
```
Comment (shazow) [inline installation-instructions/python-installation-linux.md]: This is specific to Debian-based distros.

**Comment by shazow** [comment]
Regarding the actual PR, maybe it would be best to cut out all the Python installation instructions and link to another guide which already handles all the cross-distribution differences?

**Comment by MisterRios** [comment]
Hi shazow. Thanks for the comments. My notes from the PR are to illustrate my methodology for checking that Django was indeed installed. The actual md document does not mention anything about deployment, but is only serves as an installation guide for Python. The actual HelloWebApp book includes instructions about installing on Heroku.

However, it is a good point to test on an actual machine instead of a virtual/droplet machine. I will do so today.

**Comment by MisterRios** [comment]
As to the second point-
- the guide to which the existing link goes to http://docs.python-guide.org/en/latest/starting/install/linux/ suggests first installing setup tools https://pypi.python.org/pypi/setuptools#unix-wget via wget, which is what I followed, so this document just summarizes the existing information.

Further:
- I was basing this tutorial off the existing mac tutorial and didn't proofread closer. I'll take out the brew mention.
- You are correct that pipe sudo'ing isn't the best solution. I'll amend to apt-get install python-setuptools.
- You're right that these are only Debian-based instructions. I'll have to set up a Fedora machine in the next few days and yum those.

Thanks for the feedback.

**Comment by limedaring** [comment]
Thanks @MisterRios and @shazow for contributing! Let me know when it's ready to review again. :)

**Comment by MisterRios** [comment]
Fixed the install instructions using two virtual machines.

The only note is that on Fedora, yum check-update did not require sudo, though everything else did.

**Comment by shazow** [comment]
@MisterRios Looks like a good start, thanks for doing this!

I'm still reluctant to lean too much on our own Linux instructions as there are so many distros out there, but at the same time we can probably assume some minimum amount of knowledge if the reader is using Linux.

In general, would be worth linking to @kennethreitz's [python-guide.org](http://python-guide.org) which has similar instructions and more. In fact, these kinds of Python on Linux instructions might be better off living in the Python guide project than here (and maybe linked from here).

Otherwise, this is good to merge. :)

**Comment by limedaring** [comment]
Woah totally forgot about this. Merging finally. :)

**Pull request merged**